src/Component.js

```diff
@@ -56,7 +56,7 @@ export class DebounceInput extends React.PureComponent {
     const {debounceTimeout: oldTimeout, value: oldValue} = prevProps;
     const {value: stateValue} = this.state;
 
-    if (typeof value !== 'undefined' && oldValue !== value && stateValue !== value) {
+    if (typeof value !== 'undefined' || oldValue !== value || stateValue !== value) {
```

Comment:
I changed these to OR statements, because I ran into an issue where:
1. `oldValue === value` but `stateValue !== value`, because of that I was unable to get the component to trigger `this.setState` when passing a `max` property to the input.